provider/alibabacloud/alibaba_cloud.go

```diff
@@ -661,20 +661,41 @@ func (p *AlibabaCloudProvider) updateRecords(recordMap map[string][]alidns.Recor
 
 func (p *AlibabaCloudProvider) splitDNSName(fullName string) (rr string, domain string) {
 	name := strings.TrimSuffix(fullName, ".")
-	parts := strings.Split(name, ".")
-	if len(parts) < 2 {
-		rr = name
-		domain = ""
-	} else {
-		domain = parts[len(parts)-2] + "." + parts[len(parts)-1]
-		rrIndex := strings.Index(name, domain)
-		if rrIndex < 1 {
-			rrIndex = 1
+
+	var found bool
+	domainNames, tmpErr := p.getDomainList()
+	if tmpErr != nil {
+		log.Errorf("AlibabaCloudProvider getDomainList error %v", tmpErr)
```

Comment:
Log and continue with an empty domain list seems unlikely to produce a desirable result.

Alternatives are to return the error to the caller or to get the domain list and store it in `p` before ever calling this method.

Reply:
Thanks for your review. These changes are outdated now. as suggested by stan-chen i made some changes. Can you please re-review the PR. For context on why i raised this PR please have a look at this issue https://github.com/kubernetes-sigs/external-dns/issues/3625